### PR TITLE
Drop ujson and switch to orjson

### PR DESCRIPTION
**Pull request by rumpelsepp**
Implemented as requested in this comment:
https://github.com/python-lsp/python-lsp-server/pull/508#issuecomment-1904096381

### CODE REVIEW

File: pylsp/__main__.py [hunk 1]
```diff
@@ -7,10 +7,7 @@
 import sys
 import time
 
-try:
-    import ujson as json
-except Exception:
-    import json
```
Comment (ccordoba12) [inline pylsp/__main__.py:13]: Please restore this and all other try/excepts. That's necessary for people that run this project in environments such as Pyodide.

**Review by ccordoba12 — COMMENTED** [review]
Thanks for your help with this @rumpelsepp!

Besides these changes, you also need to replace `ujson` in [python-lsp-jsonrpc](https://github.com/python-lsp/python-lsp-jsonrpc) for the switch to be complete.

File: pylsp/__main__.py (hunk 1 above)
Comment (rumpelsepp) [reply pylsp/__main__.py:13]: Done. I removed this one in this file since it is only used to load the logging config; thus not performance critical.

**Comment by rumpelsepp** [comment]
I realized there is a typing issue with this. `json.dumps()` and `orjson.dump()` return different types. I will fix this. I am short on time today.

**Comment by ccordoba12** [comment]
Closing in favor of #591. Thanks for your work on this @rumpelsepp!

**Pull request closed without merging**